### PR TITLE
[common 3] Delete currency from the project

### PR DESCRIPTION
**Pull request by Reyzis2021**
1. Task's name:
[common-3] Delete currency from the project

2. Developer's full name:
Konstantin Nyamtsy

3. Link to the PR:
https://github.com/Sunagatov/Online-Store/pull/75

4. Link to Documentation:
https://github.com/Sunagatov/Online-Store/wiki/API-Specification-for-Payment

5. Changes:
- Delete currency from entities, tables, dtos.
                                                                                                                                                                                          
6. Tests:
   1. Checked that card endpoint work as before.
   2. Checked that payment endpoint work as before.
   3.  Checked that product endpoint work as before.                                                                                                                                                                                 
                                                                                                                                                                                                                                                                              7. Test data:
  All application endpoints

8. Conclusion:
Project was built with Apache Maven successfully.

![image](https://github.com/Sunagatov/Online-Store/assets/84774469/8a32b5ad-c2e7-4ba9-8af0-25a7b9eb30e2)

### CODE REVIEW

**Review by vitaliibredun — APPROVED** [review]
Approved

**Review by yevr19 — APPROVED** [review]

File: src/main/java/com/zufar/onlinestore/payment/dto/CreatePaymentDto.java [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
+import java.math.BigDecimal;
+
 public record CreatePaymentDto(
 
         @NotBlank(message = "PaymentMethodId is the mandatory attribute")
         String paymentMethodId,
 
-        @NotNull(message = "PriceDetails is the mandatory attribute")
-        PriceDetailsDto priceDetails
+        @NotNull(message = "ItemsTotalPrice price is mandatory attribute")
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/payment/dto/CreatePaymentDto.java:13]: typo

File: src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java [hunk 2]
```diff
@@ -9,21 +9,18 @@
...
 public interface PaymentIntentConverter {
 
     @BeanMapping(ignoreByDefault = true)
     @Mapping(target = "paymentIntentId", source = "paymentIntent.id")
-    @Mapping(target = "currency", source = "paymentIntent.currency")
     @Mapping(target = "itemsTotalPrice", source = "paymentIntent.amount", qualifiedByName = {"calculateForPayment"})
     Payment toPayment(final PaymentIntent paymentIntent);
 
-    @Mapping(target = "currency", source = "createPaymentDto.priceDetails.currency")
+    @Mapping(target = "currency", constant = "usd")
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java:20]: make "usd" real constant please
`static final string variable`

**Review by Sunagatov — COMMENTED** [review]
Reviewed

File: src/main/java/com/zufar/onlinestore/payment/dto/CreatePaymentDto.java (hunk 1 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/payment/dto/CreatePaymentDto.java]: > typo

Done

File: src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java (hunk 2 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java]: > make "usd" real constant please `static final string variable

Done

File: src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
 
-    private final PaymentStatus paymentStatus;
+    private final PaymentConstants paymentConstants;
 
-    public UnexpectedPaymentStatusException(final PaymentStatus paymentStatus) {
-        super(String.format("Payment status: %s is unexpected.", paymentStatus));
-        this.paymentStatus = paymentStatus;
+    public UnexpectedPaymentStatusException(final PaymentConstants paymentConstants) {
+        super(String.format("Payment status: %s is unexpected.", paymentConstants));
+        this.paymentConstants = paymentConstants;
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java:13]: `paymentStatus` is better

File: src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
 public class UnexpectedPaymentStatusException extends RuntimeException {
 
-    private final PaymentStatus paymentStatus;
+    private final PaymentConstants paymentConstants;
 
-    public UnexpectedPaymentStatusException(final PaymentStatus paymentStatus) {
-        super(String.format("Payment status: %s is unexpected.", paymentStatus));
-        this.paymentStatus = paymentStatus;
+    public UnexpectedPaymentStatusException(final PaymentConstants paymentConstants) {
+        super(String.format("Payment status: %s is unexpected.", paymentConstants));
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java:12]: `"Payment status = %s is unexpected", paymentStatus)`

File: src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java [hunk 5]
```diff
@@ -5,7 +5,7 @@
 
 @Getter
 @RequiredArgsConstructor
-public enum PaymentStatus {
+public enum PaymentConstants {
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java:8]: PaymentStatus

File: src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java [hunk 6]
```diff
@@ -31,6 +31,8 @@ public enum PaymentStatus {
     PAYMENT_IS_SUCCEEDED("payment_intent.succeeded",
             "Payment is succeeded");
 
+    public static final String CURRENCY = "usd";
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java:34]: It is paymentStatus. paymentStatus is not good place for currency

**Review by Sunagatov — COMMENTED** [review]
Reviewed

File: src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java [hunk 7]
```diff
@@ -5,25 +5,24 @@
...
 
-    @Mapping(target = "currency", source = "createPaymentDto.priceDetails.currency")
+    @Mapping(target = "currency", constant = PaymentConstants.CURRENCY)
     @Mapping(target = "paymentMethod", source = "createPaymentDto.paymentMethodId")
     @Mapping(target = "amount",
-            source = "createPaymentDto.priceDetails.itemsTotalPrice",
+            source = "createPaymentDto.itemsTotalPrice",
             qualifiedByName = {"calculateForPaymentIntent"})
     PaymentIntentCreateParams toPaymentIntentParams(final CreatePaymentDto createPaymentDto);
+
```
Comment (Sunagatov) [inline src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java:27]: extra space

File: src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java (hunk 3 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java:13]: Done

File: src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java (hunk 4 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/payment/exception/UnexpectedPaymentStatusException.java:12]: Done

File: src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java (hunk 5 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java:8]: Done

File: src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java (hunk 6 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/payment/enums/PaymentConstants.java:34]: Done

File: src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java (hunk 7 above)
Comment (Reyzis2021) [reply src/main/java/com/zufar/onlinestore/payment/converter/PaymentIntentConverter.java:27]: done

**Review by Sunagatov — APPROVED** [review]

**Pull request merged**